src/jit/emitxarch.cpp

```diff
@@ -3765,13 +3862,15 @@ void emitter::emitIns_R_R(instruction ins, emitAttr attr, regNumber reg1, regNum
 void emitter::emitIns_R_R_I(instruction ins, emitAttr attr, regNumber reg1, regNumber reg2, int ival)
 {
     // SSE2 version requires 5 bytes and SSE4/AVX version 6 bytes
-    UNATIVE_OFFSET sz = 4;
+    UNATIVE_OFFSET sz          = 4;
+    bool           encodeAsSse = false;
```

Comment:
Why is this needed? This logic should all be controlled via `UseVEXEncoding`

Reply:
The other instructions (such as `INS_psrldq`) should be going through `emitIns_R_I`

Reply:
(for the non-vex case)

Reply:
Original logi did not work correctly for SSE2 codegen. In case of istruction with the format `ins xmm, imm` they had always being set with `id->idInsFmt(IF_RRW_RRW_CNS);` what resulted in code like `psllw xmm0, xmm0, 0` what is an invalid SSE2 instruction what resulted in CPU exception. 

Above logic ensures that RRW_CNS instructions are flagged with valid format `IF_RRW_CNS` what resolves this issue. Table based codegen should ensure this same logic.

Reply:
> In case of istruction with the format ins xmm, imm they had always being set with id->idInsFmt(IF_RRW_RRW_CNS);

The instructions should be going through an `emitIns_R_I` method for the SSE2 case. They should only go through `emitIns_R_R_I` on AVX+